<launch>

	<!-- VIDEO SOURCE -->
	<arg name="input" default="csi://0"/>
	<arg name="input_width" default="0"/>
	<arg name="input_height" default="0"/>
	<arg name="input_codec" default="unknown"/>
	<arg name="input_loop" default="0"/>

	<include file="$(find ros_deep_learning)/launch/video_source.ros1.launch">
		<arg name="input" value="$(arg input)"/>
		<arg name="input_width" value="$(arg input_width)"/>
		<arg name="input_height" value="$(arg input_height)"/>
		<arg name="input_codec" value="$(arg input_codec)"/>
		<arg name="input_loop" value="$(arg input_loop)"/>
	</include>

	<!-- SEGNET -->
	<arg name="model_name" default="fcn-resnet18-mhp-512x320"/>
	<arg name="model_path" default=""/>
	<arg name="prototxt_path" default=""/>
	<arg name="class_labels_path" default=""/>
	<arg name="class_colors_path" default=""/>
	<arg name="input_blob" default=""/>
	<arg name="output_blob" default=""/>
	<arg name="mask_filter" default="linear"/>
	<arg name="overlay_filter" default="linear"/>
	<arg name="overlay_alpha" default="180.0"/>

	<node pkg="ros_deep_learning" type="segnet" name="segnet" output="screen">
		<remap from="/segnet/image_in" to="/video_source/raw"/>
		<param name="model_name" value="$(arg model_name)"/>
		<param name="model_path" value="$(arg model_path)"/>
		<param name="prototxt_path" value="$(arg prototxt_path)"/>
		<param name="class_labels_path" value="$(arg class_labels_path)"/>
		<param name="class_colors_path" value="$(arg class_colors_path)"/>
		<param name="input_blob" value="$(arg input_blob)"/>
		<param name="output_blob" value="$(arg output_blob)"/>
		<param name="mask_filter" value="$(arg mask_filter)"/>
		<param name="overlay_filter" value="$(arg overlay_filter)"/>
		<param name="overlay_alpha" value="$(arg overlay_alpha)"/>
	</node>

	<!-- VIDEO OUTPUT -->
	<arg name="output" default="display://0"/>
	<arg name="output_codec" default="unknown"/>
	<arg name="output_bitrate" default="0"/>

	<include file="$(find ros_deep_learning)/launch/video_output.ros1.launch">
		<arg name="topic" value="/segnet/overlay"/>
		<arg name="output" value="$(arg output)"/>
		<arg name="output_codec" value="$(arg output_codec)"/>
		<arg name="output_bitrate" value="$(arg output_bitrate)"/>
	</include>

</launch>
